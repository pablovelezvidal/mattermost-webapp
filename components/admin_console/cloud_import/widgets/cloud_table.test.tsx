// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {mountWithIntl} from 'tests/helpers/intl-test-helper';

import CloudTable from './cloud_table';

const ReaactEl = (props: any) => (<p>{props.children}</p>);
const content = 'content';
const header = [
    <p key='header1'>{content}</p>,
    <ReaactEl key='header2'>{content}</ReaactEl>,
];

const values = [
    [<p key='row1-td1'>{content}</p>,
        <p key='row1-td2'>{content}</p>,
        <ReaactEl key='row1-td3'>{content}</ReaactEl>],
    [<p key='row2-td1'>{content}</p>,
        <p key='row2-td2'>{content}</p>,
        <ReaactEl key='row2-td3'>{content}</ReaactEl>],
];

describe('admin_console/cloud_table', () => {
    test('cloud table match snapshot', () => {
        const wrapper = shallow(
            <CloudTable
                header={header}
                list={values}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('cloud table displays without header', () => {
        const wrapper = mountWithIntl(
            <CloudTable list={values}/>,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
