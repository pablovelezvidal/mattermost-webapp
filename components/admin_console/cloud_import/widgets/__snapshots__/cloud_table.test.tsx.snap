// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin_console/cloud_table cloud table displays without header 1`] = `
<CloudTable
  list={
    Array [
      Array [
        <p>
          content
        </p>,
        <p>
          content
        </p>,
        <ReaactEl>
          content
        </ReaactEl>,
      ],
      Array [
        <p>
          content
        </p>,
        <p>
          content
        </p>,
        <ReaactEl>
          content
        </ReaactEl>,
      ],
    ]
  }
>
  <table
    className="CloudTable"
  >
    <thead />
    <tbody>
      <tr
        className="CloudTable__table-row"
        key="0"
      >
        <td
          key="row1-td1"
        >
          <p
            key="row1-td1"
          >
            content
          </p>
        </td>
        <td
          key="row1-td2"
        >
          <p
            key="row1-td2"
          >
            content
          </p>
        </td>
        <td
          key="row1-td3"
        >
          <ReaactEl
            key="row1-td3"
          >
            <p>
              content
            </p>
          </ReaactEl>
        </td>
      </tr>
      <tr
        className="CloudTable__table-row"
        key="1"
      >
        <td
          key="row2-td1"
        >
          <p
            key="row2-td1"
          >
            content
          </p>
        </td>
        <td
          key="row2-td2"
        >
          <p
            key="row2-td2"
          >
            content
          </p>
        </td>
        <td
          key="row2-td3"
        >
          <ReaactEl
            key="row2-td3"
          >
            <p>
              content
            </p>
          </ReaactEl>
        </td>
      </tr>
    </tbody>
  </table>
</CloudTable>
`;

exports[`admin_console/cloud_table cloud table match snapshot 1`] = `
<table
  className="CloudTable"
>
  <thead>
    <tr
      className="CloudTable__table-header"
    >
      <th
        key="0"
      >
        <p
          key="header1"
        >
          content
        </p>
      </th>
      <th
        key="1"
      >
        <ReaactEl
          key="header2"
        >
          content
        </ReaactEl>
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="CloudTable__table-row"
      key="0"
    >
      <td
        key="row1-td1"
      >
        <p
          key="row1-td1"
        >
          content
        </p>
      </td>
      <td
        key="row1-td2"
      >
        <p
          key="row1-td2"
        >
          content
        </p>
      </td>
      <td
        key="row1-td3"
      >
        <ReaactEl
          key="row1-td3"
        >
          content
        </ReaactEl>
      </td>
    </tr>
    <tr
      className="CloudTable__table-row"
      key="1"
    >
      <td
        key="row2-td1"
      >
        <p
          key="row2-td1"
        >
          content
        </p>
      </td>
      <td
        key="row2-td2"
      >
        <p
          key="row2-td2"
        >
          content
        </p>
      </td>
      <td
        key="row2-td3"
      >
        <ReaactEl
          key="row2-td3"
        >
          content
        </ReaactEl>
      </td>
    </tr>
  </tbody>
</table>
`;
